test/libsolidity/smtCheckerTests/abi/abi_encode_packed_string_literal.sol

```diff
@@ -2,29 +2,25 @@ contract C {
 	function abiencodePackedStringLiteral() public pure {
 		bytes memory b1 = abi.encodePacked("");
 		bytes memory b2 = abi.encodePacked("");
-		// should hold, but currently fails due to string literal abstraction
-		assert(b1.length == b2.length);
+		assert(b1.length == b2.length); // should hold
 
 		bytes memory b3 = abi.encodePacked(bytes(""));
-		assert(b1.length == b3.length); // should fail
+		assert(b1.length == b3.length); // should hold
```

Comment:
Should we also have assertions for cases such as the one below?

```solidity
		bytes memory a = abi.encodePacked(bytes(""));
		bytes memory b = abi.encodePacked(bytes(""));
		assert(a.length == b.length);
```

Reply:
Isn't what we have now more general?

I think I was asking about it and `""` should behave the same as `bytes("")` when fed to `abi.encode*` functions, no?

Reply:
In principle, you are right, but it feels like we could have many more tests like that, and these are not exactly cheap.
In my opinion, the basic functionality (given the same arguments, the result is the same) is already sufficiently covered.